modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -1055,11 +1055,12 @@ class Session implements ISession {
         observers.each { trace -> trace.onFlowCreate(this) }
     }
 
-    void notifyFilePublish(Path destination) {
+    void notifyFilePublish(Path source, Path destination) {
         def copy = new ArrayList<TraceObserver>(observers)
         for( TraceObserver observer : copy  ) {
             try {
                 observer.onFilePublish(destination)
+                observer.onFilePublish(source, destination)
```

Comment:
Is this the correct approach? Should I be calling both versions of `onFilePublish()`?

Reply:
Good idea! I wasn't sure if I could modify the method "stubs" in TraceObserver.